flink-kubernetes/src/main/java/org/apache/flink/kubernetes/highavailability/KubernetesLeaderElectionDriver.java

```diff
@@ -234,10 +245,21 @@ public void onError(List<KubernetesConfigMap> configMaps) {
         }
 
         @Override
-        public void handleFatalError(Throwable throwable) {
-            fatalErrorHandler.onFatalError(
-                    new LeaderElectionException(
-                            "Error while watching the ConfigMap " + configMapName, throwable));
+        public void handleError(Throwable throwable) {
+            if (throwable instanceof KubernetesTooOldResourceVersionException) {
+                synchronized (watchLock) {
+                    if (running) {
+                        LOG.info("Creating a new watch on ConfigMap {}.", configMapName);
+                        kubernetesWatch =
```

Comment:
Is it somewhere specified that `kubernetesWatch` closes automatically if an exception occurs? If not, then I think we should still call `kubernetesWatch.close` on the old instance. It might not be required right now because of some implementation details of the existing implementations, but this might change in the future.

Reply:
I do not think the `kubernetesWatch` closes automatically if an exception occurs since it could handle some exception internally(e.g. network exception). But currently we create a new watch only when the old one is closing with exception. So maybe we do not need to do the duplicated close.

Reply:
After checking the fabric8 implementation, it is harmless to close the watcher more than once. I could also add back the `kubernetesWatch.close` before creating a new one.